test/integration/tas/tas_job_test.go

```diff
@@ -0,0 +1,602 @@
+							Name: "cpu-node",
+							Labels: map[string]string{
+								"node.kubernetes.io/instance-type": "cpu-node",
+								tasRackLabel:                       "cpu-rack",
+							},
+						},
+						Status: corev1.NodeStatus{
+							Allocatable: corev1.ResourceList{
+								corev1.ResourceCPU: resource.MustParse("5"),
+							},
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "gpu-node",
+							Labels: map[string]string{
+								"node.kubernetes.io/instance-type": "gpu-node",
+								tasRackLabel:                       "gpu-rack",
+							},
+						},
+						Status: corev1.NodeStatus{
+							Allocatable: corev1.ResourceList{
+								gpuResName: resource.MustParse("4"),
+							},
+						},
+					},
+				}
+				for _, node := range nodes {
+					gomega.Expect(k8sClient.Create(ctx, &node)).Should(gomega.Succeed())
+					gomega.Expect(k8sClient.Status().Update(ctx, &node)).Should(gomega.Succeed())
+				}
+
+				topology = testing.MakeTopology("default").Levels([]string{
+					tasRackLabel,
+				}).Obj()
+				gomega.Expect(k8sClient.Create(ctx, topology)).Should(gomega.Succeed())
+
+				tasGPUFlavor = testing.MakeResourceFlavor("tas-gpu-flavor").
+					NodeLabel("node.kubernetes.io/instance-type", "gpu-node").
+					TopologyName("default").
+					Obj()
+				gomega.Expect(k8sClient.Create(ctx, tasGPUFlavor)).Should(gomega.Succeed())
+
+				tasCPUFlavor = testing.MakeResourceFlavor("tas-cpu-flavor").
+					NodeLabel("node.kubernetes.io/instance-type", "cpu-node").
+					TopologyName("default").
+					Obj()
+				gomega.Expect(k8sClient.Create(ctx, tasCPUFlavor)).Should(gomega.Succeed())
+
+				clusterQueue = testing.MakeClusterQueue("cluster-queue").
+					ResourceGroup(
+						*testing.MakeFlavorQuotas(tasGPUFlavor.Name).Resource(corev1.ResourceCPU, "1").Resource(gpuResName, "5").Obj(),
+						*testing.MakeFlavorQuotas(tasCPUFlavor.Name).Resource(corev1.ResourceCPU, "5").Resource(gpuResName, "0").Obj(),
+					).Obj()
+				gomega.Expect(k8sClient.Create(ctx, clusterQueue)).Should(gomega.Succeed())
+
+				localQueue = testing.MakeLocalQueue("local-queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+				gomega.Expect(k8sClient.Create(ctx, localQueue)).Should(gomega.Succeed())
+			})
+
+			ginkgo.AfterEach(func() {
+				gomega.Expect(util.DeleteAllJobsInNamespace(ctx, k8sClient, ns)).Should(gomega.Succeed())
+				gomega.Expect(util.DeleteWorkloadsInNamespace(ctx, k8sClient, ns)).Should(gomega.Succeed())
+				gomega.Expect(util.DeleteObject(ctx, k8sClient, localQueue)).Should(gomega.Succeed())
+				gomega.Expect(util.DeleteObject(ctx, k8sClient, topology)).Should(gomega.Succeed())
+				util.ExpectObjectToBeDeleted(ctx, k8sClient, clusterQueue, true)
+				util.ExpectObjectToBeDeleted(ctx, k8sClient, tasCPUFlavor, true)
+				util.ExpectObjectToBeDeleted(ctx, k8sClient, tasGPUFlavor, true)
+				for _, node := range nodes {
+					util.ExpectObjectToBeDeleted(ctx, k8sClient, &node, true)
```

Comment:
ditto.

"util.ExpectObjectToBeDeleted"

Reply:
Done in the follow up https://github.com/kubernetes-sigs/kueue/pull/3291